documentation/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,186 @@
+## Guide for migrating to `sdk/**/arm**` from `services/**/mgmt/**`
+
+This document is intended for users that are familiar with an older version of the Azure SDK For Go for management modules (`services/**/mgmt/**`) and wish to migrate their application to the next version of Azure resource management libraries (`sdk/**/arm**`)
+
+**For users new to the Azure SDK For Go for resource management modules, please see the [README for 'sdk/armcore`](https://github.com/Azure/azure-sdk-for-go/tree/main/sdk/armcore) and the README for every individual package.**
+
+## Table of contents
+
+* [Prerequisites](#prerequisites)
+* [General Changes](#general-changes)
+    * [Authentication](#authentication)
+    * [Customized Policy](#customized-policy)
+    * [Custom HTTP Client](#custom-http-client)
+    * [Error Handling](#error-handling)
+    * [Long Running Operations](#long-running-operations)
+    * [Pagination](#pagination)
+
+## Prerequisites
+
+Golang 1.13 or above.
+
+## General Changes
+
+The latest Azure SDK For Go for management modules is using the [Go Modules](https://github.com/golang/go/wiki/Modules) to manage the dependencies. We ship every RP as an individual module to create a more flexible user experience.
+
+The important breaking changes are listed in the following sections:
+
+### Authentication
+
+In old version (`services/**/mgmt/**`), `autorest.Authorizer` is used in authentication process.
+
+In new version (`sdk/**/arm**`), in order to provide a unified authentication based on Azure Identity for all Azure Go SDKs, the authentication mechanism has been re-designed and improved to offer a simpler interface.
+
+To the show the code snippets for the change:
+
+**In old version (`services/**/mgmt/**`)**
+
+```go
+authorizer, err := adal.NewServicePrincipalToken(oAuthToken, "<ClientId>", "<ClientSecret>", endpoint)
```

Comment:
Include example of setting `client.Authorizer`

Reply:
Done